### PR TITLE
MR-2532: State user cannot submit unlocked contract only package with rate data

### PR DESCRIPTION
**Pull request by JasonLin0991**
## Summary
[MR-2532](https://qmacbis.atlassian.net/browse/MR-2532)

In `submitHealthPlanPackage.ts` resolver, before submission, a check is done for `submissionType === CONTRACT_ONLY` and checks package for any valid rate related data using `hasAnyValidRateData` function. If both are true then `removeRatesData` function is ran to remove rate related data, including documents, before submission.

The `removeRateRelatedDocuments` function will remove any documents with `documentCategories` that are type `RATE` or `RATE_RELATED` only from the **package**. If the document `documentCategories` contains `CONTRACT` or `CONTRACT_ONLY` in addition to `RATE` or `RATE_RELATED`, then it will remove `RATE` and `RATE_RELATED` from `documentCategories` and keep `CONTRACT` or `CONTRACT_ONLY`. This ensures rate and rate supporting documents only will be removed from the package and supporting documents that are both contract and rate related will be converted to contract related only.

Removing rate related data is done on submission to preserve rate related data and documents incase the user needs to revert the submission type back to `CONTRACT_AND_RATES` before submission/resubmission. 

#### Test cases covered

**Added**
- `submitHealthPlanPackage.test.ts`
   - `'removes any rate data from CONTRACT_ONLY submissionType and submits successfully'`
   - `'does not remove any rate data from CONTRACT_AND_RATES submissionType and submits successfully'`

- `healthPlanFormData.test.ts`
   - `'Submit CONTRACT_ONLY with valid rate data: $testDescription'`

**Removed**
- `submitHealthPlanPackage.test.ts`
   - `'returns an error if there are invalid rate details fields for submission type'`
      - This was testing for valid rate data for `CONTRACT_ONLY` submission types, which is outdated, since rate data will be removed from `CONTRACT_ONLY` submission.

<strike>**Cypress**
Enabled code for rate program selection in `fillOutNewRateCertification` and `fillOutAmendmentToPriorRateCertification` commands in `stateSubmissionFormCommands.ts`. Now that `rate-program-certification` flag is enabled for all environments.</strike>

## QA guidance

- Please check this with a combination of documents and supporting documents.

### CODE REVIEW

File: services/app-web/src/common-code/healthPlanFormDataType/healthPlanFormData.ts [hunk 1]
```diff
@@ -228,16 +251,65 @@ const generateRateName = (
     return rateName
 }
 
+const removeRateRelatedDocuments = (
+    documents: SubmissionDocument[]
+): SubmissionDocument[] => {
+    const noRateDocs: SubmissionDocument[] = []
+    documents.forEach((document) => {
+        const ratesRelatedDocIndex =
+            document.documentCategories.indexOf('RATES_RELATED') === -1
```
Comment (rswerve) [inline services/app-web/src/common-code/healthPlanFormDataType/healthPlanFormData.ts:260]: Is converting to undefined serving any purpose here?  I think you could just get the index and check below if it's -1.

Comment (JasonLin0991) [reply services/app-web/src/common-code/healthPlanFormDataType/healthPlanFormData.ts:260]: Good catch! It was until, I found a bug with the code. Now it's unnecessary left over code.

**Comment by JasonLin0991** [comment]
@rswerve @haworku Now that I think of it, maybe I should disable rate program cypress commands, in `fillOutNewRateCertification` and `fillOutAmendmentToPriorRateCertification`, again and re-enable after this ticket [MR-2525](https://qmacbis.atlassian.net/browse/MR-2525)

File: services/app-web/src/common-code/healthPlanFormDataType/healthPlanFormData.ts [hunk 2]
```diff
@@ -228,16 +251,60 @@ const generateRateName = (
     return rateName
 }
 
+const removeRateRelatedDocuments = (
```
Comment (haworku) [inline services/app-web/src/common-code/healthPlanFormDataType/healthPlanFormData.ts:254]: I'm seeing some business logic in here - please add a unit test for this function.

File: services/app-web/src/common-code/healthPlanFormDataType/healthPlanFormData.test.ts [hunk 3]
```diff
@@ -541,4 +543,166 @@ describe('submission type assertions', () => {
...
+                rateAmendmentInfo: undefined,
+                rateProgramIDs: [],
+                actuaryContacts: [],
+                documents: [],
+                rateDocuments: [],
+            },
+        },
+    ]
+    test.each(contractOnlyWithValidRateData)(
+        'Submit CONTRACT_ONLY with valid rate data: $testDescription',
```
Comment (haworku) [inline services/app-web/src/common-code/healthPlanFormDataType/healthPlanFormData.test.ts:701]: Can we also test that CONTRACT_AND_RATES packages don't lose any data when removeRatesData is called?

Comment (JasonLin0991) [reply services/app-web/src/common-code/healthPlanFormDataType/healthPlanFormData.test.ts:701]: I added `'does not remove any rate data from CONTRACT_AND_RATES submissionType and submits successfully'` in `submitHealthPlanPackage.test.ts`

**Review by haworku — APPROVED** [review]

**Pull request merged**